src/main/java/com/wb/service/WeatherService.java

```diff
@@ -0,0 +1,66 @@
+package com.wb.service;
+
+import com.wb.payload.Weather;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.boot.web.client.RestTemplateBuilder;
+import org.springframework.cache.annotation.Cacheable;
+import org.springframework.scheduling.annotation.Async;
+import org.springframework.scheduling.annotation.AsyncResult;
+import org.springframework.stereotype.Service;
+import org.springframework.web.client.RestClientException;
+import org.springframework.web.client.RestTemplate;
+
+import java.util.concurrent.Future;
+
+@Service
+public class WeatherService {
+
+	private static final Logger logger = LoggerFactory.getLogger(WeatherService.class);
+	private final RestTemplate restTemplate;
+
+	@Value("${open.weather.appid}")
+	private String APP_ID;
+
+	@Value("${open.weather.app.url}")
+	private String APP_URL;
+
+	public WeatherService(RestTemplateBuilder restTemplateBuilder) {
```

Comment:
Wstrzykujesz rest template buildera a mimo to nizej deklarujesz `RestTemplate` zwyklym konstruktorem. Poważny błąd ponieważ jeśli byś mial buildera skonfigurowanego pod autentykacje to cała jego konfiguracje byś zignorował tym kodem. Użyj `restTemplateBuilder.build()`